.pylintrc

```diff
@@ -0,0 +1,90 @@
+# run arbitrary code
+extension-pkg-whitelist=
+
+# Add files or directories to the blacklist. They should be base names, not
+# paths.
+ignore=CVS
+
+# Add files or directories matching the regex patterns to the blacklist. The
+# regex matches against base names, not paths.
+ignore-patterns=
+
+# Python code to execute, usually for sys.path manipulation such as
+# pygtk.require().
+#init-hook=
+
+# Use multiple processes to speed up Pylint.
+# jobs=1
+jobs=0
+
+# List of plugins (as comma separated values of python modules names) to load,
+# usually to register additional checkers.
+load-plugins=
+
+# Pickle collected data for later comparisons.
+persistent=yes
+
+# Specify a configuration file.
+#rcfile=
+
+# Allow loading of arbitrary C extensions. Extensions are imported into the
+# active Python interpreter and may run arbitrary code.
+unsafe-load-any-extension=no
+
+
+[MESSAGES CONTROL]
+
+# Only show warnings with the listed confidence levels. Leave empty to show
+# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED
+confidence=
+
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+# disable=import-error,print-statement,parameter-unpacking,unpacking-in-except,old-raise-syntax,backtick,long-suffix,old-ne-operator,old-octal-literal,import-star-module-level,raw-checker-failed,bad-inline-option,locally-disabled,locally-enabled,file-ignored,suppressed-message,useless-suppression,deprecated-pragma,apply-builtin,basestring-builtin,buffer-builtin,cmp-builtin,coerce-builtin,execfile-builtin,file-builtin,long-builtin,raw_input-builtin,reduce-builtin,standarderror-builtin,unicode-builtin,xrange-builtin,coerce-method,delslice-method,getslice-method,setslice-method,no-absolute-import,old-division,dict-iter-method,dict-view-method,next-method-called,metaclass-assignment,indexing-exception,raising-string,reload-builtin,oct-method,hex-method,nonzero-method,cmp-method,input-builtin,round-builtin,intern-builtin,unichr-builtin,map-builtin-not-iterating,zip-builtin-not-iterating,range-builtin-not-iterating,filter-builtin-not-iterating,using-cmp-argument,eq-without-hash,div-method,idiv-method,rdiv-method,exception-message-attribute,invalid-str-codec,sys-max-int,bad-python3-import,deprecated-string-function,deprecated-str-translate-call
+disable=too-few-public-methods,
+        too-many-arguments,
+        import-error,
+        missing-docstring,
+        invalid-name
```

Comment:
The original CLI args only disable 3 rules. Let's not add new exclusions here. Only keep the ones that were set before. Ideally, this list should be empty. Check if we can have it that way.

Reply:
Could you please explain why this list should be empty? In other repositories, such as `cheroot`, I've noticed a long list. I also referred to the list here: https://github.com/adafruit/Adafruit_Learning_System_Guides/blob/main/.pylintrc.

Reply:
It's a list of rules that the linter will allow to be violated. We add linters to make sure that we don't violate the checks. There are cases, when some rules are considered unfit for a project but that should be decided and justified on per-project basis. Saying "because somebody else has this" is not a good enough justification.
When adding a linter, we should always strive to avoid any globally permitted violations and only have inline ignores when it is reasonable to make exceptions. The most straightforward to achieve when just starting a project.
Projects with legacy codebases adopting linters for the first time, may decide to temporarily allow some violations during the initial integration since it may be a huge burden to fix all occurances of the pre-existing violations.
In this case, I tend to make exceptions but then, work on fixing the violations and enabling the rules globally.

Reply:
Thank you for the comprehensive answer

Reply:
Marking as resolved.